test/test_overrides.py

```diff
@@ -461,21 +488,71 @@ def test_user_implementation_raises(self):
         with self.assertRaises(ValueError):
             quux(t1)
 
+    def test_tensor_subclass_propagation(self):
```

Comment:
It could be useful to have a comment mentioning that this test exercises the functionality described here: docs/source/notes/extending.rst#subclassing-torchtensor